core/src/main/java/com/onelogin/saml2/http/HttpRequest.java

```diff
@@ -1,150 +1,54 @@
 package com.onelogin.saml2.http;
 
-import static com.onelogin.saml2.util.Preconditions.checkNotNull;
-import static java.util.Collections.unmodifiableList;
-import static java.util.Collections.unmodifiableMap;
+import com.onelogin.saml2.util.Util;
 
-import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
-import java.util.Objects;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import org.apache.commons.lang3.StringUtils;
-
-import com.onelogin.saml2.util.Util;
-
 /**
  * Framework-agnostic representation of an HTTP request.
  *
- * @since 2.0.0
+ * @since 3.0.0
  */
-public final class HttpRequest {
-
-    public static final Map<String, List<String>> EMPTY_PARAMETERS = Collections.<String, List<String>>emptyMap();
+public interface HttpRequest {
```

Comment:
I'm OK calling this interface `SamlHttpRequest` or `OneLoginSamlHttpRequest` if the existing name is too generic.